src/Scripts/FSScriptLoader.spec.ts

```diff
@@ -0,0 +1,80 @@
+import 'reflect-metadata';
+import { DefaultVirtualScripts } from './DefaultVirtualScripts';
+import { FSScriptLoader } from './FSScriptLoader';
+import { VirtualScriptType } from './VirtualScript';
+import { logger } from '../Utils';
+import {
+  anyFunction,
+  anyString,
+  anything,
+  instance,
+  mock,
+  spy,
+  verify,
+  when
+} from 'ts-mockito';
+import { expect, use } from 'chai';
+import promisified from 'chai-as-promised';
+import fs from 'fs';
+
+use(promisified);
+
+describe('FSScriptLoader', () => {
+  const createScriptLoader = () => {
+    const mockedVirtualScripts = mock(DefaultVirtualScripts);
+    const virtualScripts = instance(mockedVirtualScripts);
+    const scriptLoader = new FSScriptLoader(virtualScripts);
+
+    return { mockedVirtualScripts, scriptLoader };
+  };
+
+  const spiedFs = spy(fs);
+  describe('load', () => {
+    it('should load scripts from paths with type local each one only once', async () => {
+      // arrange
+      const { mockedVirtualScripts, scriptLoader } = createScriptLoader();
+
+      const code = 'let a = 1;';
+
+      when(spiedFs.readFile(anyString(), anything(), anyFunction())).thenCall(
+        (_path, _opts, callback) => callback(null, code)
+      );
+
+      // act
+      const path1 = 'test.js';
+      const path2 = 'test1.js';
+
+      await scriptLoader.load({
+        [path1]: code,
+        [path2]: code
+      });
+      // assert
+      verify(
+        mockedVirtualScripts.set(path1, VirtualScriptType.LOCAL, code)
+      ).once();
+      verify(
+        mockedVirtualScripts.set(path2, VirtualScriptType.LOCAL, code)
+      ).once();
+    });
+
+    it('should log with debug level and throw if cannot read file', async () => {
+      // arrange
+      const { scriptLoader } = createScriptLoader();
+
+      when(spiedFs.readFile(anyString(), anything(), anyFunction())).thenCall(
+        (_path, _opts, callback) => callback(new Error('msg'), null)
+      );
+
+      const spiedLogger = spy(logger);
```

Comment:
Move it in to main suite (don't forget to reset spy)

Reply:
Reseting logger spy seems to break the spy. It won't longer capture calls(confirmed with just moving test that uses the spy into first position so that `reset` is never called before executing test). So, I just moved spy instantiation into beforeEach